Cargo.toml

```diff
@@ -16,10 +16,15 @@ rust-version = "1.56.0"
 [dependencies]
 vte_generate_state_changes = { version = "0.1.0", path = "vte_generate_state_changes" }
 arrayvec = { version = "0.7.2", default-features = false, optional = true }
+log = { version = "0.4.17", optional = true }
 utf8parse = { version = "0.2.0", path = "utf8parse" }
+serde = { version = "1.0.160", features = ["derive"], optional = true }
 
 [features]
+ansi = ["log"]
+serde = ["dep:serde"]
 default = ["no_std"]
+floats = []
```

Comment:
Why is this behind a floats feature?

Reply:
Anything that necessitates the use of floating point should be made optional. This is because if the crate is used use in a kernel (which is my intended use), floats are not usually available :)

Reply:
>f64 are included even in core though, right? 

Yeah they are available in core, but in a kernel you do not want to use floating point numbers because they are tedious to use. 

>I'm not aware of any crate offering explicit exclusion for floating point numbers.

If you would like to remove the feature, I could and just patch out functions that make use of floating points when I am using the crate.

Reply:
I'm mainly not entirely sure about the concrete motivation. If you have `f64`s anyway, it shouldn't matter if you have more of them, right?

These methods are completely optional, so you don't *have* to use them.

Reply:
I misinterpreted the error message from the compiler, which indicated that `f64::powf` was missing in `f64`. However, it wasn't due to the target's lack of support, but rather because it wasn't accessible in `no_std`. But still, it would be required to be gated under `no_std`.

Reply:
This should be resolved in https://github.com/alacritty/vte/pull/85/commits/18cc86b9e80a9c3a0fb2d72ce86003e04941d1b2 (now only available when `std` is available).